bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/handler/DaikinAcUnitHandler.java

```diff
@@ -44,49 +50,93 @@
+                }
+
+                updateState(DaikinBindingConstants.CHANNEL_AC_SPECIALMODE,
+                        new StringType(controlInfo.specialMode.name()));
+
+                if (controlInfo.specialMode.isUndefined()) {
+                    updateState(DaikinBindingConstants.CHANNEL_AC_SPECIALMODE_POWERFUL, UnDefType.UNDEF);
+                } else {
+                    updateState(DaikinBindingConstants.CHANNEL_AC_SPECIALMODE_POWERFUL,
+                            controlInfo.specialMode.isPowerfulActive() ? OnOffType.ON : OnOffType.OFF);
+                }
+
             }
-        }
 
-        SensorInfo sensorInfo = webTargets.getSensorInfo();
-        if (sensorInfo != null) {
-            updateTemperatureChannel(DaikinBindingConstants.CHANNEL_INDOOR_TEMP, sensorInfo.indoortemp);
+            SensorInfo sensorInfo = webTargets.getSensorInfo();
+            if (sensorInfo != null) {
+                updateTemperatureChannel(DaikinBindingConstants.CHANNEL_INDOOR_TEMP, sensorInfo.indoortemp);
+
+                updateTemperatureChannel(DaikinBindingConstants.CHANNEL_OUTDOOR_TEMP, sensorInfo.outdoortemp);
 
-            updateTemperatureChannel(DaikinBindingConstants.CHANNEL_OUTDOOR_TEMP, sensorInfo.outdoortemp);
+                if (sensorInfo.indoorhumidity.isPresent()) {
+                    updateState(DaikinBindingConstants.CHANNEL_HUMIDITY,
+                            new DecimalType(sensorInfo.indoorhumidity.get()));
+                } else {
+                    updateState(DaikinBindingConstants.CHANNEL_HUMIDITY, UnDefType.UNDEF);
+                }
+
+                if (sensorInfo.compressorfrequency.isPresent()) {
+                    updateState(DaikinBindingConstants.CHANNEL_CMP_FREQ,
+                            new DecimalType(sensorInfo.compressorfrequency.get()));
+                } else {
+                    updateState(DaikinBindingConstants.CHANNEL_CMP_FREQ, UnDefType.UNDEF);
+                }
 
-            if (sensorInfo.indoorhumidity.isPresent()) {
-                updateState(DaikinBindingConstants.CHANNEL_HUMIDITY, new DecimalType(sensorInfo.indoorhumidity.get()));
-            } else {
-                updateState(DaikinBindingConstants.CHANNEL_HUMIDITY, UnDefType.UNDEF);
+            }
+
+            try {
+                EnergyInfoYear energyInfoYear = webTargets.getEnergyInfoYear();
+
+                if (energyInfoYear.energyHeatingThisYear.isPresent()) {
+                    updateEnergyYearChannel(DaikinBindingConstants.CHANNEL_ENERGY_HEATING_CURRENTYEAR_PREFIX,
+                            energyInfoYear.energyHeatingThisYear);
+                }
+                if (energyInfoYear.energyCoolingThisYear.isPresent()) {
+                    updateEnergyYearChannel(DaikinBindingConstants.CHANNEL_ENERGY_COOLING_CURRENTYEAR_PREFIX,
+                            energyInfoYear.energyCoolingThisYear);
+                }
+            } catch (Exception e) {
```

Comment:
Please specify specific exceptions if possible. What do you expect to be thrown here anyway?

Reply:
Fixed.